### PR TITLE
Moved methods from webnode

### PR DESCRIPTION
**Pull request by jfarago**
- Added rawGenerate method to datamap.js
 - Added encryption.js file from webnode
    - Right now this is being re-exported through datamap.js. We should make a story to refactor and put the datamap.js file in the utils folder and then export both datamap and encyprtion through an index.js file. I didnt want to do it right now because I didnt want to have to make any more changes to any the other libraries depending on this.

### CODE REVIEW

File: src/datamap.js [hunk 1]
```diff
@@ -26,4 +26,48 @@ export const generate = (genesisHash, size, opts = {}) => {
...
+
+  const [dataMap, _hash] = _.reduce(
+    keys,
+    ([dataM, hash], i) => {
+      const [_obfuscatedHash, nextHash] = Encryption.hashChain(hash);
+      dataM[i] = nextHash;
+
+      return [dataM, nextHash];
+    },
+    [{}, genesisHash]
```
Comment (rfornea) [inline src/datamap.js]: I'm not positive but I think this line needs to be `[{}, forge.util.hexToBytes(genesisHash)]`

File: src/utils/encryption.js [hunk 2]
```diff
@@ -1,5 +1,126 @@
+import CryptoJS from "crypto-js";
```
Comment (rfornea) [inline src/utils/encryption.js:1]: There's some CryptoJS stuff in this file.  I think there are versions of some of these methods in webinterface that use node-forge instead.  Can you take a look?  We're moving away from CryptoJS.

Comment (rfornea) [reply src/utils/encryption.js:1]: If there are any encryption methods in both webinterface and webnode, use the versions in webinterface as they're the most up to date.  I think we updated getSalt, getPrimordialHash, obfuscate, etc., and possibly some others.

File: src/datamap.js (hunk 1 above)
Comment (jfarago) [reply src/datamap.js]: I have no idea, this is what it was in the webnode. Should i change it?

Comment (rfornea) [reply src/datamap.js]: Yes

File: src/utils/encryption.js (hunk 2 above)
Comment (jfarago) [reply src/utils/encryption.js:1]: @rfornea https://github.com/oysterprotocol/datamap-generator/commit/0a56fab3b91de5e00cb652a23ce744a36e261c82#r29171523

File: src/utils/encryption.js [hunk 3]
```diff
@@ -1,13 +1,85 @@
...
+const NONCE_LENGTH = 24;
+const TAG_LENGTH = 32;
+const TREASURE_PREFIX = _.split("Treasure: ", "")
+  .map(char => {
+    return char.charCodeAt(char).toString(16);
+  })
+  .join("");
+
+const parseEightCharsOfFilename = fileName => {
+  fileName = fileName + "________";
```
Comment (rfornea) [inline src/utils/encryption.js]: This method is still old.

Comment (jfarago) [reply src/utils/encryption.js]: Whoops, nice catch. Thank you

**Pull request merged**